compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/BallerinaSemanticModel.java

```diff
@@ -166,8 +170,11 @@ public List<Symbol> visibleSymbols(Document sourceFile, LinePosition position, D
      */
     @Override
     public Optional<Symbol> symbol(Document sourceDocument, LinePosition position) {
-        BLangCompilationUnit compilationUnit = getCompilationUnit(sourceDocument);
-        return lookupSymbol(compilationUnit, position);
+        Optional<BLangCompilationUnit> compilationUnit = getCompilationUnit(sourceDocument);
+        if (compilationUnit.isEmpty()) {
+            return Optional.empty();
```

Comment:
Can't you just return `compilationUnit`?

Reply:
Here, the idea is to return `Optional<Symbol>` not `Optional<BLangCompilationUnit>`.

Reply:
My bad, this is OK.